cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -386,17 +391,165 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
+    /**
+     * Converts the supplied JSONArray containing the load information JSONObject for each broker, into a map linking from
+     * broker ID to a map linking a more readable version of the load parameters key to their values. The load parameters
+     * that are extracted and the readable versions of the keys are dictated by the values defined in the
+     * {@link CruiseControlLoadParameters} enum.
+     *
+     * @param brokerLoadArray The JSONArray of broker load JSONObjects returned by the Cruise Control rebalance endpoint.
+     * @return A map linking from broker ID integer to a map of load parameter to value.
+     */
+    protected static Map<Integer, Map<String, Object>> extractLoadParameters(JsonArray brokerLoadArray) {
+
+        Map<Integer, Map<String, Object>> tempMap = new HashMap<>();
+
+        for (Object rawBrokerLoad : brokerLoadArray) {
+            JsonObject brokerLoad = (JsonObject) rawBrokerLoad;
+
+            Map<String, Object> brokerLoadMap = new HashMap<>();
+
+            for (CruiseControlLoadParameters intParam : CruiseControlLoadParameters.getIntegerParameters()) {
+                if (brokerLoad.containsKey(intParam.getCruiseControlKey())) {
+                    brokerLoadMap.put(intParam.getStrimziKey(), brokerLoad.getInteger(intParam.getCruiseControlKey()));
+                }
+            }
+
+            for (CruiseControlLoadParameters doubleParam : CruiseControlLoadParameters.getDoubleParameters()) {
+                if (brokerLoad.containsKey(doubleParam.getCruiseControlKey())) {
+                    brokerLoadMap.put(doubleParam.getStrimziKey(), brokerLoad.getDouble(doubleParam.getCruiseControlKey()));
+                }
+            }
+
+            int brokerID = brokerLoad.getInteger(CruiseControlRebalanceKeys.BROKER_ID.getKey());
+            tempMap.put(brokerID, brokerLoadMap);
+
+        }
+
+        return tempMap;
+
+    }
+
+    /**
+     * Converts the supplied before and after broker load arrays into a map linking from broker ID integer to a map linking
+     * from load parameter to an array of [before, after, difference]. The load paramters included in the map are dictated
+     * by the values in he {@link CruiseControlLoadParameters} enum.
+     *
+     * @param brokerLoadBeforeArray The JSONArray of broker load JSONObjects, for before the optimization proposal is applied,
+     *                              returned by the Cruise Control rebalance endpoint.
+     * @param brokerLoadAfterArray The JSONArray of broker load JSONObjects, for after the optimization proposal is applied,
+     *                             returned by the Cruise Control rebalance endpoint.
+     * @return A map linking from broker ID integer to a map of load parameter to [before, after, difference] arrays.
+     */
+    protected static Map<Integer, Map<String, Object>> createBeforeAfterLoadMap(JsonArray brokerLoadBeforeArray, JsonArray brokerLoadAfterArray) {
```

Comment:
```suggestion
    protected static Map<Integer, Map<String, Object>> parseLoadStats(JsonArray brokerLoadBeforeArray, JsonArray brokerLoadAfterArray) {
```
I think the fact it is a map is clear from the return type , maybe you can thnk of an even better name than what I suggest, but I think what it does is clearer from this that it is generating the data from what is supplied